cmd/modelcmd/controller.go

```diff
@@ -33,15 +33,44 @@ var (
 Please either create a new controller using "juju bootstrap" or connect to
 another controller that you have been given access to using "juju register".
 `)
-	// ErrNoCurrentController is returned by commands that operate on
-	// a controller if there is no current controller, no controller has been
-	// explicitly specified, and there is no default controller but there are
-	// controllers that client knows about.
-	ErrNoCurrentController = errors.New(`No selected controller.
+)
+
+// ErrNoCurrentController is returned by commands that operate on
+// a controller if there is no current controller, no controller has been
+// explicitly specified, and there is no default controller but there are
+// controllers that client knows about.
+type ErrNoCurrentController struct {
+	controllerNames []string
+}
+
+// NewNoCurrentController creates a new ErrNoCurrentController error.
+func NewNoCurrentController(names []string) ErrNoCurrentController {
+	return ErrNoCurrentController{
+		controllerNames: names,
+	}
+}
+
+func (e ErrNoCurrentController) Error() string {
+	if len(e.controllerNames) == 0 {
+		return `No selected controller.
 
 Please use "juju switch" to select a controller.
-`)
-)
+`
+	}
+
+	return fmt.Sprintf(`No selected controller.
+
+Please use "juju switch" to select from the following controllers: %s`, strings.Join(e.controllerNames, ","))
```

Comment:
I would at least add a space so that "," becomes ", ".
It might be even better to line-break them.